### DISCUSSION TITLE
How to silent print in WinUi3?

### DISCUSSION THREAD

**Discussion by MohamedSherifm** [microsoft/microsoft-ui-xaml#9669] [Q&A]
We have a kiosk application for a queueing system, which prints tickets, i want to know how can print a ticket-(designed in xaml this is the approach we were using in WPF)- using mostly POS printer with no specific brand. without showing any print dialogs.

**Comment by castorix** [comment]
I have no POS printer to test, but there was this answer from _Xavier Xie-MSFT_ for UWP which could work with WinUI 3 :

"
Besides, there's a specific scenario: The POS printer API. You could print without a Print Dialog. This POS printer API can be used for all the printers which support [ESC/POS]. You could call the PrintLine(string) method with the content and execute it with ExecuteAsync method.

The [POS Printer sample](https://github.com/microsoft/Windows-universal-samples/tree/main/Samples/PosPrinter) is for your reference. I just mentioned this API is for the specific scenario.
"

(otherwise, there are Win32 Printer APIs but I don't know if they work in a kiosk application...)

**Reply by MohamedSherifm** [reply]
Is there any Documentations on how to use the Win32 Printer APIs in WinUi?

**Reply by DarranRowe** [reply]
The general WinUI 3 environment is just a desktop application running WinUI 3 as a library. So the documentation for desktop application applies.
